lib/rules/no-deprecated-delete-set.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * @author Wayne Zhang
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+const utils = require('../utils')
+const { findVariable } = require('@eslint-community/eslint-utils')
+
+const deprecatedApis = ['set', 'delete']
+const deprecatedImportApis = new Set(['set', 'del'])
+const deprecatedDollarApis = new Set(deprecatedApis.map((item) => `$${item}`))
```

Comment:
Let's simplify this a bit:

```suggestion
const deprecatedImportApis = new Set(['set', 'del'])
const deprecatedDollarApis = new Set(['$set', '$delete'])
```

Reply:
Well, since we need `this.$delete()`, `Vue.delete`, and `del`,  the three `Set` should each contain `$delete`, `delete`, and `del`, respectively.